server/datastore/mysql/policies.go

```diff
@@ -1663,3 +1663,38 @@ func (ds *Datastore) GetTeamHostsPolicyMemberships(
 
 	return hosts, nil
 }
+
+// GetPoliciesBySoftwareTitleID returns the policies that are associated with a set of software titles.
+func (ds *Datastore) getPoliciesBySoftwareTitleIDs(
+	ctx context.Context,
+	softwareTitleIDs []uint,
+	teamID *uint,
+) ([]fleet.AutomaticInstallPolicy, error) {
+	query := `
+	SELECT
+		p.id AS id,
+		p.name AS name,
+		st.id AS software_title_id
+	FROM policies p
+	JOIN software_installers si ON p.software_installer_id = si.id
+	JOIN software_titles st ON si.title_id = st.id
+	WHERE st.id IN (?) AND p.team_id = ?
+	GROUP BY software_title_id, id;
+`
+
+	var tmID uint
+	if teamID != nil {
+		tmID = *teamID
+	}
+
+	query, args, err := sqlx.In(query, softwareTitleIDs, tmID)
```

Comment:
The `sqlx.In` unfortunately returns an invalid statement when the `IN (?)` results in 0 elements. This should probably be checked first and `return nil, nil` if `len(softwareTitleIDs) == 0`.